ui/test/unit/registration/controllers/patientCommonController.spec.js

```diff
@@ -361,6 +361,41 @@ it('checks that the confirmation popup is not prompted on the Registration secon
             expect(sections.additionalPatientInformation.canShow).toBeTruthy();
         })
     })
+    
+    describe("Call function for duplicate patients", function () {
+        it("should call function for checking duplicate patient", function() {
+         scope.patient.systemIdentier = 'BAH203007';
+         scope.patient.givenName = 'Dhruv';
+         scope.patient.familyName = 'Bhardwaj';
+         scope.patient.gender = 'M';
+         scope.patient.birthDate = '1993-12-18T00:00:00.000Z';
+         scope.patient.PERSON_ATTRIBUTE_TYPE_PHONE_NUMBER = '987654321';
+         scope.patient.address = 'BAYA HADJIDA';
+         expect(scope.patient.systemIdentier).toBe('BAH203007');
+         expect(scope.patient.givenName).toBe('Dhruv');
+         expect(scope.patient.familyName).toBe('Bhardwaj');
+         expect(scope.patient.gender).toBe('M');
+         expect(scope.patient.birthDate).toBe('1993-12-18T00:00:00.000Z');
+         expect(scope.patient.PERSON_ATTRIBUTE_TYPE_PHONE_NUMBER).toBe('987654321');
+         expect(scope.patient.address).toBe('BAYA HADJIDA');
+         scope.checkDuplicatePatients = jasmine.createSpy("checkDuplicatePatients");
+         scope.checkDuplicatePatients();
```

Comment:
@prashantsable1991 one objective of this test is to verify if the duplicate check is being triggered, therefore we cannot manually call scope.checkDuplicatePatients() from within the test. There is a watcher on scope.patient.address that triggers the call checkDupliicatePatient (see https://github.com/jembi/cameroon-openmrs-module-bahmniapps/blob/COM-165/ui/app/registration/controllers/patientCommonController.js#L220), therefore the duplicate check method should be called because of the address being set on line 373